test-harness/specs/parameters-ac1.oas2.txt

```diff
@@ -0,0 +1,28 @@
+====test====
+When I send a request to an operation
+And the operation has a required path segment
+And in the request I sent that segment is missing
+I should get a 400 response back
+And the payload of the response should be in application/problem+json content-type
+====spec====
+swagger: '2.0'
+paths:
+  /pet/a/{b}:
+    get:
+      parameters:
+        - in: path
+          name: b
+          required: true
+          type: number
+      responses:
+        '200':
+          description: ok
+          schema:
+            type: string
+====server====
+mock -p 4010
+====command====
+curl -sIXGET http://127.0.0.1:4010/pet/a
```

Comment:
What is `-sIXGET ` doing? This looks like a mistake?

Reply:
I think is:

`-I`: show *only* the headers, and not the body
`-s`: do not show progress messages
`-X` specify the verb

Reply:
basically I could just send a `HEAD`, but thought it would be more natural to keep `get` in the spec

Reply:
I am impressed that curl is able to parse that GET out of there, and not confuse it for another switch. We talked yesterday about how we don't need `-X GET` ever because it is default.

Reply:
BTW a lot of UNIX software is able to understand "glued" parameters

`-I test` = `-Itest`. I'm curious about why such thing.

Reply:
Yep my last message was saying two things:

1. cool and I learned a thing
2. we dont need -X GET ever because it is the default

Reply:
Apparently when using -s the verb is mandatory, otherwise curl sends a POST. I've restored the command.

Reply:
What an interface 🤦🏻‍♂️ Thank you :)